perf: enhance useSSE hook to manage modified events and fetch sources efficiently

BenElferink opened the pull request:
emergency-ticket id: EMR-766

BenElferink committed:
refactor: update OverviewHeader component and improve SSE error handling

- Simplified OverviewHeader component by consolidating left and right elements using useMemo for better performance.
- Enhanced error handling in useSSE hook to manage reconnection attempts with exponential backoff.
- Removed unused imports and cleaned up code in various hooks for improved readability and maintainability.
- Updated status store to include optional priority messages and left icon support.

BenElferink committed:
Merge branch 'main' into ben/plat-598-copy-to-oss

BenElferink committed:
refactor: enhance useSSE hook with improved state management and debounce logic

- Introduced constants for debounce timing and maximum events per fetch.
- Refactored state management using refs for last modified event IDs and timestamps.
- Added a reset function for cleaning up event references.
- Updated logic to handle fetching sources based on modified event timing and count.

BenElferink committed:
refactor: enhance useSSE and useSourceCRUD hooks with status management

- Integrated status store into useSSE and useSourceCRUD hooks for improved user feedback.
- Added clearStatusMessage function to reset status messages in useSSE.
- Updated notification handling logic to ensure proper state management during source modifications.
- Improved code readability by removing unused imports and consolidating logic.

BenElferink committed:
refactor: update debounce timing in useSSE hook for improved performance

- Increased debounce timing from 5000ms to 8000ms to reduce unnecessary fetch calls.
- This change aims to enhance the efficiency of the useSSE hook in managing notifications.

BenElferink committed:
refactor: update OverviewHeader component to use OdigosLogoText

- Replaced OdigosCentralLogoText with OdigosLogoText in the OverviewHeader component for consistency with design updates.
- This change enhances the visual representation of the header while maintaining existing functionality.

alonbraymok committed:
chore: alon backend debounce

alonbraymok committed:
chore: wip

BenElferink committed:
Merge pull request #38 from alonkeyval/ben/plat-598-copy-to-oss

Ben/plat 598 copy to oss

BenElferink committed:
refactor: enhance useSSE and useSourceCRUD hooks for improved event handling and state management

- Refactored useSSE hook to utilize a single event reference for managing timestamps and intervals, improving clarity and performance.
- Updated notification handling logic to ensure proper state management during source modifications, including handling added and deleted sources.
- Removed unused functions and consolidated logic in useSourceCRUD for better maintainability.
- Improved the preparation of source payloads by separating add and delete operations for clearer state updates.

BenElferink committed:
refactor: update instrumentation config watchers and notification handling

- Adjusted debounce duration comments in instrumentation config watchers for clarity.
- Enhanced notification messages in useSSE hook to handle undefined values, ensuring accurate feedback during source creation and deletion.
- These changes improve the consistency and reliability of event handling across the application.

BenElferink committed:
refactor: swap notification event types in useSSE hook for accurate handling

- Changed the event handling logic in the useSSE hook to correctly differentiate between ADDED and MODIFIED events.
- This adjustment ensures that the appropriate actions are taken based on the type of notification received, improving the reliability of event processing.

BenElferink committed:
refactor: standardize debounce duration and improve event handling in instrumentation config watchers

- Updated comments in instrumentation config watchers to reflect the correct debounce duration reference.
- Modified debounce behavior in the useSSE hook to ensure consistent handling of event intervals.
- Enhanced notification logic to provide clearer feedback during source operations, improving overall user experience.

BenElferink committed:
refactor: update notification messages in useSSE hook for clarity

- Changed status messages in the useSSE hook to provide clearer feedback during source creation and instrumentation processes.
- Updated messages to enhance user experience by specifying actions and waiting times, ensuring better communication of the system's state.

BenElferink committed:
Merge branch 'main' into ben/plat-598-copy-to-oss

BenElferink committed:
Merge branch 'main' into ben/plat-598-copy-to-oss

BenElferink committed:
Merge branch 'main' into ben/plat-598-copy-to-oss

alonkeyval commented on frontend/webapp/hooks/notification/useSSE.ts:
```diff
@@ -44,58 +112,70 @@ export const useSSE = () => {
+              setInstrumentCount('sourcesCreated', totalCreated);
+
+              handleEventInterval(EventTypes.ADDED, () => {
+                const statusMessage2 = 'Instrumentating sources, please wait a moment...';
```
Instrumentating -> Instrumenting

alonkeyval commented on frontend/webapp/hooks/notification/useSSE.ts:
```diff
@@ -20,20 +19,89 @@ enum CrdTypes {
+
+    // if last message was over `XXX` seconds ago, fetch all sources...
+    // the interval runs a timestamp check every 1 second - once the condition is met, the interval is cleared.
+    eventsRef.current[eventType]!.interval = setInterval(() => {
```
is'nt it better to use setTimeout here? why we need to check every second if we only need to check after 5 sec?

BenElferink committed:
refactor: enhance event handling and debounce logic in useSSE hook

- Updated the useSSE hook to replace interval-based event handling with timeout-based handling for improved responsiveness.
- Renamed functions and variables for clarity, ensuring better understanding of their roles in the event processing flow.
- Adjusted debounce behavior to provide more accurate event triggering, enhancing user experience during source operations.

BenElferink committed:
Merge branch 'main' into ben/plat-598-copy-to-oss
